app/models/enterprise.rb

```diff
@@ -187,8 +177,25 @@ def self.distinct_count
     count(distinct: true)
   end
 
+  def contact
+    manager = EnterpriseRole.find_by_enterprise_id_and_receives_notifications(self.id, true)
+    return owner if manager.blank?
+    contact = Spree::User.find(manager.user_id)
+    contact || owner
+  end
```

Comment:
Why is it preferable to query `EnterpriseRole` rather than using it as a join table to ask for a `User`?

````ruby
def contact
  contact = users.where(enterprise_roles: { receives_notifications: true} ).first
  contact || owner
end